cartography/intel/aws/iam.py

```diff
@@ -364,11 +384,125 @@ def load_user_access_keys(neo4j_session, user_access_keys, aws_update_tag):
+    neo4j_session.run(
+        injest_policy,
+        PolicyId=policy_id,
+        PolicyName=policy_name,
+        PolicyType=policy_type,
+        PrincipalArn=principal_arn,
+        aws_update_tag=aws_update_tag,
+    ).consume()
+
+
+@timeit
+def load_policy_statements(neo4j_session, policy_id, policy_name, statements, aws_update_tag):
+    injest_policy_statement = """
+        MATCH (policy:AWSPolicy{id: {PolicyId}})
+        WITH policy
+        UNWIND {Statements} as statement_data
+        MERGE (statement:AWSPolicyStatement{id: statement_data.id})
+        SET
+        statement.effect = statement_data.Effect,
+        statement.action = statement_data.Action,
+        statement.notaction = statement_data.NotAction,
+        statement.resource = statement_data.Resource,
+        statement.notresource = statement_data.NotResource,
+        statement.condition = statement_data.Condition,
+        statement.sid = statement_data.Sid,
+        statement.lastupdated = {aws_update_tag}
+        MERGE (policy)-[r:STATEMENT]->(statement)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {aws_update_tag}
+        """
+    neo4j_session.run(
+        injest_policy_statement,
+        PolicyId=policy_id,
+        PolicyName=policy_name,
+        Statements=statements,
+        aws_update_tag=aws_update_tag,
+    ).consume()
+
+
+@timeit
+def load_policy_data(neo4j_session, policy_map, aws_update_tag):
+    for principal_arn, policy_list in policy_map.items():
+        logger.debug(f"Syncing IAM inline policies for principal {principal_arn}")
+        for policy in policy_list:
+            name = policy.name
+            policy_id = f"{principal_arn}/inline_policy/{name}"
+            statements = _generate_policy_statements(policy.policy_document["Statement"], policy_id)
+            load_policy(neo4j_session, policy_id, name, "inline", principal_arn, aws_update_tag)
+            load_policy_statements(neo4j_session, policy_id, name, statements, aws_update_tag)
+
+
+@timeit
+def load_managed_policy_data(neo4j_session, policy_map, aws_update_tag):
+    for principal_arn, policy_list in policy_map.items():
+        for policy in policy_list:
+            name = policy.policy_name
+            policy_id = policy.arn
+            statements = _generate_policy_statements(policy.default_version.document["Statement"], policy_id)
+            load_policy(neo4j_session, policy_id, name, "managed", principal_arn, aws_update_tag)
+            load_policy_statements(neo4j_session, policy_id, name, statements, aws_update_tag)
+
+
 @timeit
 def sync_users(neo4j_session, boto3_session, current_aws_account_id, aws_update_tag, common_job_parameters):
     logger.debug("Syncing IAM users for account '%s'.", current_aws_account_id)
     data = get_user_list_data(boto3_session)
     load_users(neo4j_session, data['Users'], current_aws_account_id, aws_update_tag)
+
+    policy_data = get_user_policy_data(boto3_session, data['Users'])
```

Comment:
To keep things in smaller testable chunks and for consistency, I think this should be separated out to its own sync() function.

Reply:
They all require the same user list. I can break out the sub parts into three functions, but the testability would be the same.